### PR TITLE
[WEB-559] [WEB-818] react upgrade/component library

### PR DESCRIPTION
**Pull request by krystophv**
[WEB-818] - merge react upgrade into develop to provide access to [WEB-559] component library on main development branch

[WEB-818]: https://tidepool.atlassian.net/browse/WEB-818
[WEB-559]: https://tidepool.atlassian.net/browse/WEB-559

### CODE REVIEW

File: test/components/common/stat/Stat.test.js [hunk 1]
```diff
@@ -1762,7 +1762,7 @@ describe('Stat', () => {
         index: 1,
       });
 
-      sinon.assert.callCount(setStateSpy, 1);
+      // sinon.assert.callCount(setStateSpy, 1);
```
Comment (cbwebdevelopment) [inline test/components/common/stat/Stat.test.js:1765]:
```suggestion
```

File: test/components/common/stat/Stat.test.js [hunk 2]
```diff
@@ -1782,7 +1782,7 @@ describe('Stat', () => {
 
       instance.setChartTitle();
 
-      sinon.assert.callCount(setStateSpy, 1);
+      // sinon.assert.callCount(setStateSpy, 1);
```
Comment (cbwebdevelopment) [inline test/components/common/stat/Stat.test.js:1785]:
```suggestion
```

File: test/components/common/stat/Stat.test.js [hunk 3]
```diff
@@ -1799,7 +1799,7 @@ describe('Stat', () => {
...
```
Comment (cbwebdevelopment) [inline test/components/common/stat/Stat.test.js:1802]:
```suggestion
```

File: src/components/settings/common/PumpSettingsContainer.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import _ from 'lodash';
```
[the remark this answers is not in the document]
Comment (krystophv) [reply src/components/settings/common/PumpSettingsContainer.js:1]: Yeah, the codemods did a lot of that sort of thing. Tried to find all the instances with some global finds, but inevitably some slip through. Thanks for the catch

**Comment by krystophv** [comment]
@cbwebdevelopment I'm good with that workaround. Since it seems to only affect the unit tests and not the user experience, I think we can proceed with that for now, but it does serve as an indicator that there's some fiddly bits with relying on the `UNSAFE` deprecations and that we'll have some work ahead of us when it comes time to convert over (it seems they're committed to supporting them at least through 17.x, so we can effectively kick the can down the road for a while)

**Pull request merged**